NetSDK/Services/Cache/Classes/SegmentCache.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Splitio.Services.Cache.Interfaces;
+using Splitio.Services.SegmentFetcher.Classes;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Splitio.Services.Cache.Classes
+{
+    public class SegmentCache : ISegmentCache
+    {
+        private ConcurrentDictionary<string, SelfRefreshingSegment> segments;
+
+        public void AddToSegment(string segmentName, HashSet<string> segmentKeys)
+        {
+            SelfRefreshingSegment segment;
+            if (segments.TryGetValue(segmentName, out segment))
+            {
+                segment.AddKeys(segmentKeys);
```

Comment:
is `SelfRefreshingSegment` thread-safe? Just to confirm we are not going to mess-up things.